frontend/src/metabase/components/Timeline.jsx

```diff
@@ -0,0 +1,95 @@
+import React, { useMemo } from "react";
+import PropTypes from "prop-types";
+import cx from "classnames";
+import _ from "underscore";
+import styled from "styled-components";
+import moment from "moment";
+
+import { color } from "metabase/lib/colors";
+
+import Icon from "metabase/components/Icon";
+
+const TimelineContainer = styled.div`
+  position: relative;
+  margin-left: ${props => props.leftShift}px;
+  margin-bottom: ${props => props.bottomShift}px;
+`;
+
+const TimelineItem = styled.div`
+  transform: translateX(-${props => props.leftShift}px);
+  white-space: pre-line;
+`;
+
+// shift the border down slightly so that it doesn't appear above the top-most icon
+const Border = styled.div`
+  position: absolute;
+  top: ${props => props.borderShift}px;
+  left: 0;
+  right: 0;
+  bottom: -${props => props.borderShift}px;
+  border-left: 1px solid ${color("border")};
+`;
+
+const Timeline = ({ className, items = [], renderFooter }) => {
+  const iconSize = 20;
+  const halfIconSize = iconSize / 2;
+
+  const sortedFormattedItems = useMemo(() => {
+    return items
+      .sort((a, b) => b.timestamp - a.timestamp)
+      .map(item => {
+        return {
+          ...item,
+          formattedTimestamp: moment(item.timestamp).fromNow(),
+        };
+      });
+  }, [items]);
+
+  return (
+    <TimelineContainer
+      leftShift={halfIconSize}
```

Comment:
What does the shift mean?

Reply:
It's a bad name, but it felt a _little_ better/more informative than a `marginLeft={halfIconSize}` prop. The design (a left border with overlaid icons to give a sense of connection between timeline events) requires me to shift the icons over by half their width. In order to avoid a scenario in the future where there is something to the left of a `Timeline` interfering with the timeline UI, I've also forced there to be a left margin of `halfIconSize`

Reply:
Happy to take renaming suggestions, however 🙂

Reply:
Got it, yeah I can't think of anything better at the moment.